src/shell.zig

```diff
@@ -596,12 +596,33 @@ pub fn spawn_options(
     return child;
 }
 
+pub fn zig_exe_alloc(shell: Shell, allocator: std.mem.Allocator) ![]const u8 {
```

Comment:
For simplicity, let's rather do this during `Shell.create` and store the result as a field, like how we do for `project_root`. 

I debated whether we should keep shell.zig as an independent re-usable utility, or whether to allow dependency on TigerBeetle repo specifically, and I figured that a bounded amount of dependency is probably better.k

As a bonus, this'll tell us eagarly if we miss the environmental variable in our build.zig!

Reply:
Ah, that is much better! We even already have the `EnvMap` in `Shell.create`, which simplifies the error handling.

Though, we do miss the environment variable often -- e.g. `build.zig` _itself_ uses `Shell`. It just doesn't end up being needed in those cases.